src/renate/benchmark/datasets/vision_datasets.py

```diff
@@ -471,3 +472,106 @@ def setup(self) -> None:
+            original source.
+        src_object_name: The folder path in the s3 bucket.
+        scenario: One of ``ni``, ``nc``, ``nic``, ``nicv2_79``, ``nicv2_196`` and ``nicv2_391``.
+            This is different from the usage of scenario elsewhere in Renate.
+        data_id: One of the several data batches dependent on scenario
+    """
+
+    md5s = {
+        "core50_128x128.zip": "745f3373fed08d69343f1058ee559e13",
+        "paths.pkl": "b568f86998849184df3ec3465290f1b0",
+        "LUP.pkl": "33afc26faa460aca98739137fdfa606e",
+        "labels.pkl": "281c95774306a2196f4505f22fd60ab1",
+    }
+    dataset_stats = {
+        "Core50": dict(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225]),
+    }
+
+    def __init__(
+        self,
+        data_path: Union[Path, str],
+        src_bucket: Optional[str] = None,
+        src_object_name: Optional[str] = None,
+        scenario: Literal["ni", "nc", "nic", "nicv2_79", "nicv2_196", "nicv2_391"] = "ni",
+        data_id: int = 0,
+        val_size: float = defaults.VALIDATION_SIZE,
+        seed: int = defaults.SEED,
+    ) -> None:
+        super().__init__(
+            data_path=data_path,
+            src_bucket=src_bucket,
+            src_object_name=src_object_name,
+            data_id=data_id,
+            val_size=val_size,
+            seed=seed,
+        )
+        self._dataset_name = "core50"
+        self._image_source = "core50_128x128"
+        self._scenario = scenario
+        self._complete_data_path = os.path.join(
+            self._data_path, self._dataset_name, self._image_source
+        )
+
+    def prepare_data(self) -> None:
+        """Download the CORE50 dataset and supporting files and set paths."""
+        if not self._verify_file(f"{self._image_source}.zip"):
+            download_and_unzip_file(
+                self._dataset_name,
+                self._data_path,
+                self._src_bucket,
+                self._src_object_name,
+                "http://bias.csr.unibo.it/maltoni/download/core50/",
+                f"{self._image_source}.zip",
+            )
+        for file_name in [
+            "paths.pkl",
+            "LUP.pkl",
+            "labels.pkl",
+        ]:
+            if not self._verify_file(file_name):
+                download_file(
+                    self._dataset_name,
+                    self._data_path,
+                    self._src_bucket,
+                    self._src_object_name,
+                    "https://vlomonaco.github.io/core50/data/",
+                    file_name,
+                )
+        with open(os.path.join(self._data_path, self._dataset_name, "paths.pkl"), "rb") as f:
+            self._paths = pickle.load(f)
+
+        with open(os.path.join(self._data_path, self._dataset_name, "LUP.pkl"), "rb") as f:
+            self._LUP = pickle.load(f)
+
+        with open(os.path.join(self._data_path, self._dataset_name, "labels.pkl"), "rb") as f:
+            self._labels = pickle.load(f)
```

Comment:
this should be in `setup`. otherwise, it will not be in memory when we access the attributes in `setup`.